airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py

```diff
@@ -281,8 +380,34 @@ def upload_metadata_to_gcs(bucket_name: str, metadata_file_path: Path, validator
     Returns:
         Tuple[bool, str]: Whether the metadata file was uploaded and its blob id.
     """
-    icon_file_path = metadata_file_path.parent / ICON_FILE_NAME
-    metadata_file_path = _apply_modifications_to_metadata_file(metadata_file_path, validator_opts)
+    # Get our working directory and temp directories
+    working_directory = metadata_file_path.parent
+    temp_directory = working_directory / "temp"
+    temp_directory.mkdir(exist_ok=True)
+
+    # Declare paths to the files that may be uploaded to GCS
+    yaml_manifest_file_path = working_directory / "manifest.yaml"
+    components_py_file_path = working_directory / "components.py"
+    python_components_zip_file_path = temp_directory / "python_components.zip"
+    python_components_zip_sha256_file_path = temp_directory / "python_components.zip.sha256"
+
+    components_zip_sha256: str | None = None
+    if components_py_file_path.exists():
```

Comment:
We will need to update airbyte_ci Metadata step as well to include this file https://github.com/airbytehq/airbyte/blob/07-30-feat-cat_add_manifest_support/airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/metadata/pipeline.py/#L82-L85

Reply:
@bnchrch - I don't think I follow this comment.

Reply:
@aaronsteers Sorry! The link is broken
https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/metadata/pipeline.py/#L82-L85

Essentially we'll have to mount the components file to the metadata upload container. 

Actually lets just mount the whole connector.

Ill make that change for you!

Reply:
@aaronsteers Just pushed a commit with the change.

cc: @alafanechere for your thoughts on this as the tooling wizard.

Essentially instead of always updating the ci steps to include more and more individual files for validation, I just mounted the entire connector.

This will come with a publish speed penalty.

But perhaps if that speed penalty becomes a penalty we look at 
1. mounting the connector once
2. using a shared volume to prevent re-mounts?

I dont know if thats possible in dagger yet though